core/runtime/binaryen/runtime_external_interface.cpp

```diff
@@ -111,16 +111,15 @@ namespace {
  * @param name method name
  * @return result of method invocation
  */
-#define CALL_HOST_API_FUNC(name)                                         \
-  do {                                                                   \
-    if (import->base == #name) {                                         \
-      checkArguments(import->base.c_str(),                               \
-                     hostApiFuncArgSize<&host_api::HostApi ::name>(),    \
-                     arguments.size());                                  \
-      return callHostApiFunc<&host_api::HostApi ::name>(host_api_.get(), \
-                                                        arguments);      \
-    }                                                                    \
-  } while (false)  // hack to make macro call look natural by ending with ';'
+#define REGISTER_HOST_API_FUNC(name)                                        \
+  imports_[#name] = [this](wasm::Function *import,                          \
+                           wasm::LiteralList &arguments) -> wasm::Literal { \
+    checkArguments(import->base.c_str(),                                    \
+                   hostApiFuncArgSize<&host_api::HostApi ::name>(),         \
+                   arguments.size());                                       \
+    return callHostApiFunc<&host_api::HostApi ::name>(host_api_.get(),      \
+                                                      arguments);           \
+  };
```

Comment:
Please preserve this to avoid warnings from some compilers on useless duplication of `;`
```suggestion
  }  // hack to make macro call look natural by ending with ';'
```